src/Tribe/Tickets.php

```diff
@@ -1371,85 +1355,39 @@ public function front_end_tickets_form_in_content( $content ) {
 		}
 
 		/**
-		 * Returns a (possibly empty) string used to direct users to the new user
-		 * registration form when appropriate.
+		 * Indicates if the user must be logged in in order to obtain tickets.
 		 *
-		 * Typically used when a requirement for users to be logged in to access the ticket
-		 * form is in effect and where registrations are allowed, but where the user is
-		 * currently logged out.
+		 * This should be regarded as an abstract method to be overridden by subclasses:
+		 * the reason it is not formally declared as abstract is to avoid breakages upon
+		 * update (for example, where Event Tickets is updated first but a dependent plugin
+		 * not yet implementing the abstract method remains at an earlier version).
 		 *
-		 * @return string
+		 * @return bool
 		 */
-		protected function login_or_register_advice() {
-			// We only want to display this text once (this will fire as many times as there are
-			// active ticket modules) and then only if user registration is enabled
-			if ( self::$have_displayed_reg_link ) {
-				return '';
-			}
-
-			self::$have_displayed_reg_link = true;
-
-			// We should include a redirect query for a good UX and because some plugins
-			// like Community Events can ban users from the admin environment (let's not
-			// leave them stranded)
-			$login_link = add_query_arg(
-				'redirect_to',
-				rawurlencode( get_permalink() ),
-				get_site_url( null, 'wp-login.php' )
-			);
-
-			$login_links = '<a href="' . esc_url( $login_link ) . '">' . __( 'Login', 'event-tickets' ) . '</a>';
-
-			if ( get_option( 'users_can_register' ) ) {
-				$login_links .= ' | ' . wp_register( '', '', false );
-			}
-
-			/**
-			 * Modify the text that displays when users are required to be logged in to see
-			 * the ticket form but where they are currently logged out and where registration
-			 * of new user accounts is enabled.
-			 *
-			 * @param string $register_help_text
-			 */
-			return apply_filters( 'event_tickets_register_to_see_ticket_form_message', sprintf(
-				__( 'You must be logged in to obtain tickets. %s', 'event-tickets' ),
-				$login_links
-			) );
+		protected function login_required() {
+			return false;
 		}
 
 		/**
-		 * Indicates if the ticket form is enabled.
+		 * Provides a URL that can be used to direct users to the login form.
 		 *
-		 * Generally this will return true, but there may be special occasions such
-		 * as access being denied to logged out users and similar where it returns
-		 * false.
-		 *
-		 * @return bool
+		 * @return string
 		 */
-		protected function form_is_enabled() {
-			$enabled = true;
+		public static function get_login_link() {
```

Comment:
I wonder if this method should be named `get_login_url()` instead? It returns a URL rather than a link.